server/src/main/java/org/opensearch/action/fieldcaps/TransportFieldCapabilitiesIndexAction.java

```diff
@@ -266,7 +266,8 @@ private ShardRouting nextRoutingOrNull(Exception failure) {
             if (shardsIt.size() == 0 || shardIndex >= shardsIt.size()) {
                 return null;
             }
-            ShardRouting next = FailAwareWeightedRouting.getInstance().findNext(shardsIt.get(shardIndex), clusterService.state(), failure);
+            ShardRouting next = FailAwareWeightedRouting.getInstance()
+                .findNext(shardsIt.get(shardIndex), clusterService.state(), failure, () -> ++shardIndex);
```

Comment:
Nitpick, but I think you could replace `() -> ++shardIndex` with `this::moveToNextShard` and it would probably be better object-oriented practice and maybe read a bit more clearly.